internal/strategies/flexible_rollout.go

```diff
@@ -0,0 +1,78 @@
+package strategies
+
+import (
+	"math/rand"
+	"strconv"
+
+	"github.com/Unleash/unleash-client-go/v3/context"
+	"github.com/Unleash/unleash-client-go/v3/strategy"
+)
+
+type stickiness string
+
+const (
+	defaultStickiness   stickiness = "default"
+	userIDStickiness    stickiness = "userId"
+	sessionIDStickiness stickiness = "sessionId"
+	randomStickiness    stickiness = "random"
+)
+
+type flexibleRolloutStrategy struct {
+	random *rand.Rand
+}
+
+// NewFlexibleRolloutStrategy creates a new instance of the flexible rollout strategy.
+func NewFlexibleRolloutStrategy() *flexibleRolloutStrategy {
+	s := &flexibleRolloutStrategy{
+		random: newRand(),
+	}
+	return s
+}
+
+func (s flexibleRolloutStrategy) Name() string {
+	return "flexibleRollout"
+}
+
+func (s flexibleRolloutStrategy) randomNumber() string {
+	return strconv.Itoa(s.random.Intn(100) + 1)
```

Comment:
There is a data race here. It's not safe to call methods on rand.Rand from more than one goroutine concurrently. If a user of the module calls Client.IsEnabled from multiple goroutines concurrently then we might end up in this function in more than one goroutine concurrently. There is a similar data race in gradualRolloutRandomStrategy.

Reply:
Good catch, pushed a fix now.